\doxysection{Référence de l\textquotesingle{}espace de nommage numpy.\+\_\+core.\+\_\+ufunc\+\_\+config}
\hypertarget{namespacenumpy_1_1__core_1_1__ufunc__config}{}\label{namespacenumpy_1_1__core_1_1__ufunc__config}\index{numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}}
\doxysubsubsection*{Classes}
\begin{DoxyCompactItemize}
\item 
class \mbox{\hyperlink{classnumpy_1_1__core_1_1__ufunc__config_1_1__unspecified}{\+\_\+unspecified}}
\item 
class \mbox{\hyperlink{classnumpy_1_1__core_1_1__ufunc__config_1_1errstate}{errstate}}
\end{DoxyCompactItemize}
\doxysubsubsection*{Fonctions}
\begin{DoxyCompactItemize}
\item 
\mbox{\hyperlink{namespacenumpy_1_1__core_1_1__ufunc__config_a9f4457c62a8551915ce249ef6057606f}{seterr}} (\mbox{\hyperlink{namespacenumpy_1_1__core_1_1fromnumeric_a6ae68c91210062722ef04204c10c4b5a}{all}}=None, divide=None, over=None, under=None, invalid=None)
\item 
\mbox{\hyperlink{namespacenumpy_1_1__core_1_1__ufunc__config_aa4c39abd9823c53ec931895ceca07610}{geterr}} ()
\item 
\mbox{\hyperlink{namespacenumpy_1_1__core_1_1__ufunc__config_ad80f2a868e0065784a73dc952f6cdd04}{setbufsize}} (\mbox{\hyperlink{namespacenumpy_1_1__core_1_1fromnumeric_a1bcb53966eb8c8f3e05fcee1848b27c1}{size}})
\item 
\mbox{\hyperlink{namespacenumpy_1_1__core_1_1__ufunc__config_a9d2b8c7c70a922adcc2b9c6b19094fc4}{getbufsize}} ()
\item 
\mbox{\hyperlink{namespacenumpy_1_1__core_1_1__ufunc__config_a9e05ca7f7366a632634fa91e7b697184}{seterrcall}} (func)
\item 
\mbox{\hyperlink{namespacenumpy_1_1__core_1_1__ufunc__config_a7538363c4c2c60cedb00710e2919efa0}{geterrcall}} ()
\end{DoxyCompactItemize}
\doxysubsubsection*{Variables}
\begin{DoxyCompactItemize}
\item 
\mbox{\hyperlink{namespacenumpy_1_1__core_1_1__ufunc__config_a2a5911ebcd10e32acbf13b79524d1654}{\+\_\+\+Unspecified}} = \mbox{\hyperlink{classnumpy_1_1__core_1_1__ufunc__config_1_1__unspecified}{\+\_\+unspecified}}()
\end{DoxyCompactItemize}


\doxysubsection{Description détaillée}
\begin{DoxyVerb}Functions for changing global ufunc configuration

This provides helpers which wrap `_get_extobj_dict` and `_make_extobj`, and
`_extobj_contextvar` from umath.
\end{DoxyVerb}
 

\doxysubsection{Documentation des fonctions}
\Hypertarget{namespacenumpy_1_1__core_1_1__ufunc__config_a9d2b8c7c70a922adcc2b9c6b19094fc4}\label{namespacenumpy_1_1__core_1_1__ufunc__config_a9d2b8c7c70a922adcc2b9c6b19094fc4} 
\index{numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}!getbufsize@{getbufsize}}
\index{getbufsize@{getbufsize}!numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}}
\doxysubsubsection{\texorpdfstring{getbufsize()}{getbufsize()}}
{\footnotesize\ttfamily numpy.\+\_\+core.\+\_\+ufunc\+\_\+config.\+getbufsize (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})}

\begin{DoxyVerb}Return the size of the buffer used in ufuncs.

Returns
-------
getbufsize : int
    Size of ufunc buffer in bytes.

Examples
--------
>>> import numpy as np
>>> np.getbufsize()
8192\end{DoxyVerb}
 \Hypertarget{namespacenumpy_1_1__core_1_1__ufunc__config_aa4c39abd9823c53ec931895ceca07610}\label{namespacenumpy_1_1__core_1_1__ufunc__config_aa4c39abd9823c53ec931895ceca07610} 
\index{numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}!geterr@{geterr}}
\index{geterr@{geterr}!numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}}
\doxysubsubsection{\texorpdfstring{geterr()}{geterr()}}
{\footnotesize\ttfamily numpy.\+\_\+core.\+\_\+ufunc\+\_\+config.\+geterr (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})}

\begin{DoxyVerb}Get the current way of handling floating-point errors.

Returns
-------
res : dict
    A dictionary with keys "divide", "over", "under", and "invalid",
    whose values are from the strings "ignore", "print", "log", "warn",
    "raise", and "call". The keys represent possible floating-point
    exceptions, and the values define how these exceptions are handled.

See Also
--------
geterrcall, seterr, seterrcall

Notes
-----
For complete documentation of the types of floating-point exceptions and
treatment options, see `seterr`.

Examples
--------
>>> import numpy as np
>>> np.geterr()
{'divide': 'warn', 'over': 'warn', 'under': 'ignore', 'invalid': 'warn'}
>>> np.arange(3.) / np.arange(3.)  # doctest: +SKIP
array([nan,  1.,  1.])
RuntimeWarning: invalid value encountered in divide

>>> oldsettings = np.seterr(all='warn', invalid='raise')
>>> np.geterr()
{'divide': 'warn', 'over': 'warn', 'under': 'warn', 'invalid': 'raise'}
>>> np.arange(3.) / np.arange(3.)
Traceback (most recent call last):
  ...
FloatingPointError: invalid value encountered in divide
>>> oldsettings = np.seterr(**oldsettings)  # restore original\end{DoxyVerb}
 \Hypertarget{namespacenumpy_1_1__core_1_1__ufunc__config_a7538363c4c2c60cedb00710e2919efa0}\label{namespacenumpy_1_1__core_1_1__ufunc__config_a7538363c4c2c60cedb00710e2919efa0} 
\index{numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}!geterrcall@{geterrcall}}
\index{geterrcall@{geterrcall}!numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}}
\doxysubsubsection{\texorpdfstring{geterrcall()}{geterrcall()}}
{\footnotesize\ttfamily numpy.\+\_\+core.\+\_\+ufunc\+\_\+config.\+geterrcall (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})}

\begin{DoxyVerb}Return the current callback function used on floating-point errors.

When the error handling for a floating-point error (one of "divide",
"over", "under", or "invalid") is set to 'call' or 'log', the function
that is called or the log instance that is written to is returned by
`geterrcall`. This function or log instance has been set with
`seterrcall`.

Returns
-------
errobj : callable, log instance or None
    The current error handler. If no handler was set through `seterrcall`,
    ``None`` is returned.

See Also
--------
seterrcall, seterr, geterr

Notes
-----
For complete documentation of the types of floating-point exceptions and
treatment options, see `seterr`.

Examples
--------
>>> import numpy as np
>>> np.geterrcall()  # we did not yet set a handler, returns None

>>> orig_settings = np.seterr(all='call')
>>> def err_handler(type, flag):
...     print("Floating point error (%s), with flag %s" % (type, flag))
>>> old_handler = np.seterrcall(err_handler)
>>> np.array([1, 2, 3]) / 0.0
Floating point error (divide by zero), with flag 1
array([inf, inf, inf])

>>> cur_handler = np.geterrcall()
>>> cur_handler is err_handler
True
>>> old_settings = np.seterr(**orig_settings)  # restore original
>>> old_handler = np.seterrcall(None)  # restore original\end{DoxyVerb}
 \Hypertarget{namespacenumpy_1_1__core_1_1__ufunc__config_ad80f2a868e0065784a73dc952f6cdd04}\label{namespacenumpy_1_1__core_1_1__ufunc__config_ad80f2a868e0065784a73dc952f6cdd04} 
\index{numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}!setbufsize@{setbufsize}}
\index{setbufsize@{setbufsize}!numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}}
\doxysubsubsection{\texorpdfstring{setbufsize()}{setbufsize()}}
{\footnotesize\ttfamily numpy.\+\_\+core.\+\_\+ufunc\+\_\+config.\+setbufsize (\begin{DoxyParamCaption}\item[{}]{size }\end{DoxyParamCaption})}

\begin{DoxyVerb}Set the size of the buffer used in ufuncs.

.. versionchanged:: 2.0
    The scope of setting the buffer is tied to the `numpy.errstate`
    context.  Exiting a ``with errstate():`` will also restore the bufsize.

Parameters
----------
size : int
    Size of buffer.

Returns
-------
bufsize : int
    Previous size of ufunc buffer in bytes.

Examples
--------
When exiting a `numpy.errstate` context manager the bufsize is restored:

>>> import numpy as np
>>> with np.errstate():
...     np.setbufsize(4096)
...     print(np.getbufsize())
...
8192
4096
>>> np.getbufsize()
8192\end{DoxyVerb}
 \Hypertarget{namespacenumpy_1_1__core_1_1__ufunc__config_a9f4457c62a8551915ce249ef6057606f}\label{namespacenumpy_1_1__core_1_1__ufunc__config_a9f4457c62a8551915ce249ef6057606f} 
\index{numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}!seterr@{seterr}}
\index{seterr@{seterr}!numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}}
\doxysubsubsection{\texorpdfstring{seterr()}{seterr()}}
{\footnotesize\ttfamily numpy.\+\_\+core.\+\_\+ufunc\+\_\+config.\+seterr (\begin{DoxyParamCaption}\item[{}]{all = {\ttfamily None},  }\item[{}]{divide = {\ttfamily None},  }\item[{}]{over = {\ttfamily None},  }\item[{}]{under = {\ttfamily None},  }\item[{}]{invalid = {\ttfamily None} }\end{DoxyParamCaption})}

\begin{DoxyVerb}Set how floating-point errors are handled.

Note that operations on integer scalar types (such as `int16`) are
handled like floating point, and are affected by these settings.

Parameters
----------
all : {'ignore', 'warn', 'raise', 'call', 'print', 'log'}, optional
    Set treatment for all types of floating-point errors at once:

    - ignore: Take no action when the exception occurs.
    - warn: Print a :exc:`RuntimeWarning` (via the Python `warnings`
      module).
    - raise: Raise a :exc:`FloatingPointError`.
    - call: Call a function specified using the `seterrcall` function.
    - print: Print a warning directly to ``stdout``.
    - log: Record error in a Log object specified by `seterrcall`.

    The default is not to change the current behavior.
divide : {'ignore', 'warn', 'raise', 'call', 'print', 'log'}, optional
    Treatment for division by zero.
over : {'ignore', 'warn', 'raise', 'call', 'print', 'log'}, optional
    Treatment for floating-point overflow.
under : {'ignore', 'warn', 'raise', 'call', 'print', 'log'}, optional
    Treatment for floating-point underflow.
invalid : {'ignore', 'warn', 'raise', 'call', 'print', 'log'}, optional
    Treatment for invalid floating-point operation.

Returns
-------
old_settings : dict
    Dictionary containing the old settings.

See also
--------
seterrcall : Set a callback function for the 'call' mode.
geterr, geterrcall, errstate

Notes
-----
The floating-point exceptions are defined in the IEEE 754 standard [1]_:

- Division by zero: infinite result obtained from finite numbers.
- Overflow: result too large to be expressed.
- Underflow: result so close to zero that some precision
  was lost.
- Invalid operation: result is not an expressible number, typically
  indicates that a NaN was produced.

.. [1] https://en.wikipedia.org/wiki/IEEE_754

Examples
--------
>>> import numpy as np
>>> orig_settings = np.seterr(all='ignore')  # seterr to known value
>>> np.int16(32000) * np.int16(3)
np.int16(30464)
>>> np.seterr(over='raise')
{'divide': 'ignore', 'over': 'ignore', 'under': 'ignore', 'invalid': 'ignore'}
>>> old_settings = np.seterr(all='warn', over='raise')
>>> np.int16(32000) * np.int16(3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FloatingPointError: overflow encountered in scalar multiply

>>> old_settings = np.seterr(all='print')
>>> np.geterr()
{'divide': 'print', 'over': 'print', 'under': 'print', 'invalid': 'print'}
>>> np.int16(32000) * np.int16(3)
np.int16(30464)
>>> np.seterr(**orig_settings)  # restore original
{'divide': 'print', 'over': 'print', 'under': 'print', 'invalid': 'print'}\end{DoxyVerb}
 \Hypertarget{namespacenumpy_1_1__core_1_1__ufunc__config_a9e05ca7f7366a632634fa91e7b697184}\label{namespacenumpy_1_1__core_1_1__ufunc__config_a9e05ca7f7366a632634fa91e7b697184} 
\index{numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}!seterrcall@{seterrcall}}
\index{seterrcall@{seterrcall}!numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}}
\doxysubsubsection{\texorpdfstring{seterrcall()}{seterrcall()}}
{\footnotesize\ttfamily numpy.\+\_\+core.\+\_\+ufunc\+\_\+config.\+seterrcall (\begin{DoxyParamCaption}\item[{}]{func }\end{DoxyParamCaption})}

\begin{DoxyVerb}Set the floating-point error callback function or log object.

There are two ways to capture floating-point error messages.  The first
is to set the error-handler to 'call', using `seterr`.  Then, set
the function to call using this function.

The second is to set the error-handler to 'log', using `seterr`.
Floating-point errors then trigger a call to the 'write' method of
the provided object.

Parameters
----------
func : callable f(err, flag) or object with write method
    Function to call upon floating-point errors ('call'-mode) or
    object whose 'write' method is used to log such message ('log'-mode).

    The call function takes two arguments. The first is a string describing
    the type of error (such as "divide by zero", "overflow", "underflow",
    or "invalid value"), and the second is the status flag.  The flag is a
    byte, whose four least-significant bits indicate the type of error, one
    of "divide", "over", "under", "invalid"::

      [0 0 0 0 divide over under invalid]

    In other words, ``flags = divide + 2*over + 4*under + 8*invalid``.

    If an object is provided, its write method should take one argument,
    a string.

Returns
-------
h : callable, log instance or None
    The old error handler.

See Also
--------
seterr, geterr, geterrcall

Examples
--------
Callback upon error:

>>> def err_handler(type, flag):
...     print("Floating point error (%s), with flag %s" % (type, flag))
...

>>> import numpy as np

>>> orig_handler = np.seterrcall(err_handler)
>>> orig_err = np.seterr(all='call')

>>> np.array([1, 2, 3]) / 0.0
Floating point error (divide by zero), with flag 1
array([inf, inf, inf])

>>> np.seterrcall(orig_handler)
<function err_handler at 0x...>
>>> np.seterr(**orig_err)
{'divide': 'call', 'over': 'call', 'under': 'call', 'invalid': 'call'}

Log error message:

>>> class Log:
...     def write(self, msg):
...         print("LOG: %s" % msg)
...

>>> log = Log()
>>> saved_handler = np.seterrcall(log)
>>> save_err = np.seterr(all='log')

>>> np.array([1, 2, 3]) / 0.0
LOG: Warning: divide by zero encountered in divide
array([inf, inf, inf])

>>> np.seterrcall(orig_handler)
<numpy.Log object at 0x...>
>>> np.seterr(**orig_err)
{'divide': 'log', 'over': 'log', 'under': 'log', 'invalid': 'log'}\end{DoxyVerb}
 

\doxysubsection{Documentation des variables}
\Hypertarget{namespacenumpy_1_1__core_1_1__ufunc__config_a2a5911ebcd10e32acbf13b79524d1654}\label{namespacenumpy_1_1__core_1_1__ufunc__config_a2a5911ebcd10e32acbf13b79524d1654} 
\index{numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}!\_Unspecified@{\_Unspecified}}
\index{\_Unspecified@{\_Unspecified}!numpy.\_core.\_ufunc\_config@{numpy.\_core.\_ufunc\_config}}
\doxysubsubsection{\texorpdfstring{\_Unspecified}{\_Unspecified}}
{\footnotesize\ttfamily numpy.\+\_\+core.\+\_\+ufunc\+\_\+config.\+\_\+\+Unspecified = \mbox{\hyperlink{classnumpy_1_1__core_1_1__ufunc__config_1_1__unspecified}{\+\_\+unspecified}}()\hspace{0.3cm}{\ttfamily [protected]}}

